### PR TITLE
A non-smoke test version of the end-to-end test

### PR DESCRIPTION
**Pull request by rebkwok**

### CODE REVIEW

**Review by benbc — CHANGES_REQUESTED** [review]
I feel like there must be a slightly cleaner way of doing this. At the moment the understanding about how to handle the two different ways of running the extraction is spread out between the test functions (where `method` and `args` are specified) and `run_test()` (where the logic for the results is).

Could we instead pass in a "runner" callable that takes care of both aspects? Something a bit like this (not checked it properly):

```
def test_extracts_data_from_sql_server_integration_tests(load_study, load data, database):
    study = load_study("end_to_end_tests")
    
    def runner():
        main(str(study.path), "mock", database.host_url())
        return study.path / "outputs
        
    run_test(study, load_data, runner)
```

File: tests/test_end_to_end.py [hunk 1]
```diff
@@ -27,18 +28,30 @@ def run(study):
...
 
 def assert_results_equivalent(actual_results, expected_results):
-    with open(actual_results / "cohort.csv") as actual_file:
-        with open(expected_results) as expected_file:
-            actual_data = list(csv.DictReader(actual_file))
-            expected_data = list(csv.DictReader(expected_file))
+    actual_results_file = actual_results / "cohort.csv"
+    assert actual_results_file.read_text(
+        encoding="utf-8"
+    ) == expected_results.read_text(encoding="utf-8")
```
[the remark this answers is not in the document]
Comment (rebkwok) [reply tests/test_end_to_end.py:34]: Maybe it's useful if/when the output files get more complex. I can change it back if you prefer

File: tests/conftest.py [hunk 2]
```diff
@@ -16,24 +17,29 @@
...
     def read_dir(path):
         return Study(path)
 
-    return read_dir
+    yield read_dir
+    # Clean up any output directories and files created by tests
+    fixtures_path = Path(__file__).parent.absolute() / "fixtures"
+    output_dirs = fixtures_path.glob("*/outputs")
+    for output_dir in output_dirs:
+        shutil.rmtree(output_dir)
```
[the remark this answers is not in the document]
Comment (rebkwok) [reply tests/conftest.py:42]: The temporary directory is only created in the container.  When main is called directly, it uses the fixtures folder.  I'll try making it also use the copying-to-temp-dir

[the remark this answers is not in the document]
Comment (rebkwok) [reply tests/conftest.py]: Done.  Not sure if I prefer it this way more or not

**Review by benbc — APPROVED** [review]

**Pull request merged**